node/src/Pos/Block/Network/Retrieval.hs

```diff
@@ -89,75 +89,105 @@ retrievalWorkerImpl SendActions {..} =
         queue        <- view (lensOf @BlockRetrievalQueueTag)
         recHeaderVar <- view (lensOf @RecoveryHeaderTag)
         logDebug "Waiting on the block queue or recovery header var"
+        -- Reading the queue is a priority, because it sets the recovery
+        -- variable in case the header is classified as alternative. So if the
+        -- queue contains lots of headers after a long delay, we'll first
+        -- iterate over them and set recovery variable to the latest one, and
+        -- only then we'll do recovery.
         thingToDoNext <- atomically $ do
             mbQueuedHeadersChunk <- tryReadTBQueue queue
             mbRecHeader <- tryReadTMVar recHeaderVar
             case (mbQueuedHeadersChunk, mbRecHeader) of
                 (Nothing, Nothing) -> retry
+                -- Dispatch the task
                 (Just (nodeId, task), _) ->
-                    pure (handleBlockRetrievalFromQueue nodeId task)
+                    pure (handleBlockRetrieval nodeId task)
+                -- No tasks & the recovery header is set => do the recovery
                 (_, Just (nodeId, rHeader))  ->
-                    pure (handleHeadersRecovery nodeId rHeader)
+                    pure (handleRecoveryWithHandler nodeId rHeader)
         thingToDoNext
         mainLoop
     mainLoopE e = do
         -- REPORT:ERROR 'reportOrLogE' in block retrieval worker.
-        reportOrLogE "retrievalWorker: error caught " e
-        delay (30 & sec)
+        reportOrLogE "retrievalWorker mainLoopE: error caught " e
+        delay (sec 1)
```

Comment:
This seems to change the delay from 30secs to just 1 -- is this what we want? (Just checking) I think the answer is "yes" as our main aim is to speed up recovery.